### PR TITLE
resource_manager: introduce the resource unit model and config

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: ref #5851.

### What is changed and how does it work?

```commit-message
Introduce the resource unit (RU) model and related config.
```

### Check List

Tests

- Unit test
- Integration test

Related changes

- This PR is split from #5811.

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]
lgtm.

File: pkg/mcs/resource_manager/client/config.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+const (
+	defaultReadBaseCost     = 1
+	defaultReadCostPerByte  = 1. / 1024 / 1024
+	defaultWriteBaseCost    = 5
```
[the remark this answers is not in the document]
Comment (JmPotato) [reply pkg/mcs/resource_manager/client/config.go]: Make sense. Let's keep it a smaller value first. We can adjust it after some tests.

File: pkg/mcs/resource_manager/client/model.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	}
+}
+
+// AfterKVRequest ...
+func (kc *KVCalculator) AfterKVRequest(consumption *rmpb.Consumption, req RequestInfo, res ResponseInfo) {
+	if req.IsWrite() {
+		kvCPUMs := float64(res.KVCPUMs())
+		consumption.TotalCpuTimeMs += kvCPUMs
+		consumption.WRU += float64(kc.WriteCPUMsCost) * kvCPUMs
+	} else {
```
Comment (CabinfeverB) [inline pkg/mcs/resource_manager/client/model.go:87]: Write request also can make read flow, do we need add these flow into RRU?

File: pkg/mcs/resource_manager/client/model.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	*Config
+}
+
+// func newSQLCalculator(cfg *Config) *SQLCalculator {
+// 	return &SQLCalculator{Config: cfg}
+// }
+
+// Trickle ...
+// TODO: calculate the SQL CPU cost and related resource consumption.
+func (dsc *SQLCalculator) Trickle(ctx context.Context, resource map[rmpb.ResourceType]float64, ru map[rmpb.RequestUnitType]float64) {
```
Comment (CabinfeverB) [inline pkg/mcs/resource_manager/client/model.go:105]: use `consumption *rmpb.Consumption` as param

File: pkg/mcs/resource_manager/client/model.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+// BeforeKVRequest ...
+func (dsc *SQLCalculator) BeforeKVRequest(resource map[rmpb.ResourceType]float64, ru map[rmpb.RequestUnitType]float64, req RequestInfo) {
```
Comment (CabinfeverB) [inline pkg/mcs/resource_manager/client/model.go:109]: ditto

File: pkg/mcs/resource_manager/client/model.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+// AfterKVRequest ...
+func (dsc *SQLCalculator) AfterKVRequest(resource map[rmpb.ResourceType]float64, ru map[rmpb.RequestUnitType]float64, req RequestInfo, res ResponseInfo) {
```
Comment (CabinfeverB) [inline pkg/mcs/resource_manager/client/model.go:113]: ditto

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**